nym-vpn-desktop/src-tauri/src/network.rs

```diff
@@ -1,9 +1,62 @@
-use tracing::debug;
+use std::path::PathBuf;
+
+use anyhow::{anyhow, Context, Result};
+use nym_vpn_lib::nym_config;
+use tokio::fs::{read, try_exists};
+use tracing::{debug, error, info, instrument};
+
+#[instrument]
+pub async fn setup_network_env(use_sandbox: bool, env_config_file: &Option<PathBuf>) -> Result<()> {
+    if use_sandbox {
+        info!("network environment: sandbox");
+        // TODO: instead bundle a sandbox env file we can use
+        setup_sandbox_environment();
+        return Ok(());
+    }
+
+    if let Some(file) = env_config_file {
+        debug!("provided env_config_file: {}", file.display());
+
+        let context = format!(
+            "app config, failed to read the provided `env_config_file`: `{}`",
+            file.display()
+        );
+
+        // check if the file exists / is accessible
+        if !(try_exists(file)
```

Comment:
Suggestion: usually the check if a file exists happens when reading it, since it might change anyway between this call and the read call